crates/task-impls/src/response.rs

```diff
@@ -84,14 +84,22 @@ impl<TYPES: NodeType, V: Versions> NetworkResponseState<TYPES, V> {
                     match event.as_ref() {
                         HotShotEvent::VidRequestRecv(request, sender) => {
                             let cur_epoch = self.consensus.read().await.cur_epoch();
+                            let next_epoch = cur_epoch.map(|epoch| epoch + 1);
+                            let target_epoch = if self.valid_sender(sender, cur_epoch).await {
+                                cur_epoch
+                            } else if self.valid_sender(sender, next_epoch).await {
+                                next_epoch
+                            } else {
+                                // The sender neither belongs to the current nor to the next epoch.
+                                continue;
+                            };
                             // Verify request is valid
-                            if !self.valid_sender(sender, cur_epoch).await
-                                || !valid_signature::<TYPES>(request, sender)
-                            {
+                            if !valid_signature::<TYPES>(request, sender) {
                                 continue;
                             }
-                            if let Some(proposal) =
-                                self.get_or_calc_vid_share(request.view, sender).await
+                            if let Some(proposal) = self
+                                .get_or_calc_vid_share(request.view, target_epoch, sender)
+                                .await
```

Comment:
Hmm just thinking through this, do we need more changes to disperse the shares to each epoch or is that already done.  Basically for the transition block there is 2 sets of vid shares for both the outgoing and upcoming epoch.

Maybe that's already completed?